website/docs/guides/upgrade.html.markdown

```diff
@@ -0,0 +1,158 @@
+to make sure that the acknowledged warnings are from the correct execution, and to enforce execution of the 
+`nsxt_upgrade_precheck_acknowledge` resource after preparation is complete.
+
+The `nsxt_upgrade_prepare_ready` data source verifies that the preparation phase of the upgrade was concluded 
+successfully and enable the execution of the upgrade. 
+
+```hcl
+resource "nsxt_upgrade_precheck_acknowledge" "test" {
+  precheck_ids   = ["backupOperationCheck", "pUBCheck"]
+  target_version = nsxt_upgrade_prepare.test.target_version
+}
+
+data "nsxt_upgrade_prepare_ready" "ready" {
+  upgrade_prepare_id = nsxt_upgrade_prepare.prepare_res.id
+  depends_on         = [nsxt_upgrade_precheck_acknowledge.test]
+}
+```
+
+### Running the upgrade
+
+In order to configure and execute upgrade of NSXT edges, hosts, and managers, use `nsxt_upgrade_run` resource. 
+
+This resource will start with configuration of upgrade unit groups and upgrade plan settings for `EDGE`and `HOST` 
+components. Host component groups may be created within this resource with custom host lists and attributes, or it can 
+use the predefined groups which NSX provides.
+
+Following the configuration of the edge, host groups, it will execute the upgrade for each component. If there are 
+either `EDGE` or `HOST` groups disabled from the upgrade, the corresponding component will be paused after enabled 
+groups get upgraded. At the same time, `MP` upgrade won't be processed, because that requires `EDGE` and `HOST` to be 
+fully upgraded. Refer to the exposed attribute for current upgrade state details. For example, check `upgrade_plan`
+for current upgrade plan, which also includes the plan not specified in this resource and `state` for upgrade status of 
+each component and UpgradeUnitGroups in the component. For more details, please check NSX admin guide.
+
+In the example below, host_group `HostWithDifferentSettings` will use different settings than the defaulted ones. The 
+group configuration is initiated with the `nsxt_upgrade_run` while the other groups in this example are predefined and 
+consumed using the data sources `nsxt_edge_upgrade_group` and `nsxt_host_upgrade_group`.
+
+```hcl
+resource "nsxt_upgrade_run" "run" {
+  upgrade_prepare_ready_id = data.nsxt_upgrade_prepare_ready.ready.id
+
+  edge_group {
+    id      = data.nsxt_edge_upgrade_group.eg1.id
+    enabled = false
+  }
+
+  edge_group {
+    id      = data.nsxt_edge_upgrade_group.eg2.id
+    enabled = true
+  }
+
+  host_group {
+    id       = data.nsxt_host_upgrade_group.hg1.id
+    parallel = true
+  }
+  host_group {
+    display_name = "HostWithDifferentSettings"
+    parallel     = false
+    hosts        = ["375c260d-3196-487c-b5eb-046de958f6af"]
+  }
+
+  edge_upgrade_setting {
+    parallel           = true
+    post_upgrade_check = true
+  }
+
+  host_upgrade_setting {
+    parallel           = true
+    post_upgrade_check = true
+  }
+
+  host_group {
+    display_name = "TEST123"
```

Comment:
did you add this group for a different purpose, it's not mentioned in the text above, but it looks like you could have written:

host_group `HostWithDifferentSettings` and `TEST123` will use different settings than the defaulted ones

Reply:
It's just a copy-paste error, but maybe I'll reuse it as you suggested.